### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by ekohl**
See individual commits. Based on https://github.com/dLobatog/npm2rpm/pull/32 but taken a step further.

### CODE REVIEW

File: default.n2r [hunk 1]
```diff
@@ -32,18 +30,16 @@ $BUILD
...
 mkdir -p %{buildroot}%{nodejs_sitelib}/%{npm_name}
 $COPYFILES
-# If any binaries are included, symlink them to bindir here
 $BINSNIPPET
 
-%nodejs_symlink_deps
+$SYMLINK
 
 %clean
 rm -rf %{buildroot} %{npm_cache_dir}
```
Comment (ehelms) [inline default.n2r:36]: This line should only exist if its bundled and defining npm_cache_dir.

**Comment by ehelms** [comment]
Small comment but otherwise looks like a good adaptation of my changes.

File: default.n2r (hunk 1 above)
Comment (ekohl) [reply default.n2r:36]: Should the buildroot still be removed in ```%clean```? for unbundled? Or is that directory actually inside the buildroot and should the space be removed?

Comment (ehelms) [reply default.n2r:36]: Standard practice is to include a buildroot clean command. I think the npm_cache_dir cleanup is relative to whether it's used.

Comment (ekohl) [reply default.n2r:36]: I think the default clean is ```rm -rf '%{buildroot}'``` and if we'd place ```%{npm_cache_dir}``` inside the buildroot then it'd automatically clean it up. Do you know if this is correct or am I making things up?

**Review by dLobatog — APPROVED** [review]
@ekohl I think the clean command is fine, npm_cache_dir may not be in buildroot, it's always defined as:

```
 %define npm_cache_dir /tmp/npm_cache_%{name}-%{version}-%{release}` shows
```

it's true that on non-bundled packages it may be good to just avoid any references to `npm_cache_dir` as @ehelms said, it's just an useless var. i think I'm going to go with merging, feel free to fix that in another PR

Thanks!

**Pull request merged**